front/app/components/PageForm/index.tsx

```diff
@@ -25,57 +46,154 @@ export interface FormValues {
   local_page_files: UploadFile[] | null;
 }
 
-export interface Props {
-  hideTitle?: boolean;
-  hideSlugInput?: boolean;
+type PageFormProps = {
+  onSubmit: (formValues: FormValues) => void | Promise<void>;
+  defaultValues?: FormValues;
   pageId: string | null;
-}
-
-export function validatePageForm(
-  appConfigurationLocales: Locale[],
-  existingSlugs?: Set<string>,
-  currentSlug?: string
-) {
-  return function ({
-    title_multiloc,
-    body_multiloc,
-    slug,
-  }: FormValues): FormikErrors<FormValues> {
-    const errors: FormikErrors<FormValues> = {};
-
-    errors.title_multiloc = validateMultiloc(
-      title_multiloc,
-      appConfigurationLocales
-    );
-    errors.body_multiloc = validateMultiloc(
-      body_multiloc,
-      appConfigurationLocales
-    );
-    errors.slug = validateSlug(slug, existingSlugs, currentSlug);
-
-    return removeUndefined(errors);
-  };
-}
+  hideSlugInput?: boolean;
+} & InjectedIntlProps;
 
 const PageForm = ({
-  values,
-  errors,
-  hideTitle,
-  hideSlugInput,
+  intl: { formatMessage },
+  onSubmit,
+  defaultValues,
   pageId,
-  ...props
-}: FormikProps<FormValues> & Props) => (
-  <BasePageForm values={values} errors={errors} {...props}>
-    {!hideTitle && <PageTitleField error={errors.title_multiloc} />}
+  hideSlugInput,
+}: PageFormProps) => {
+  const { add } = useToast();
```

Comment:
We've introduced a toast mechanism to give feedback to the user on success / failure of form submission.